// @uipath/robot/dist/models/jobStatus.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.JobStatus = void 0;
/**
 * Job status model containing details of the executing job.
 */
var JobStatus = /** @class */ (function () {
    /**
     *
     * @param state Current state of the job
     * @param status Current status of the job
     * @param events Workflow events sent by the job
     * @param output Output of the job if any
     * @param exception Exception from the job if any
     */
    function JobStatus(state, status, events, output, exception) {
        this.state = state;
        this.status = status;
        this.events = events;
        this.output = output;
        this.exception = exception;
        this.events = events || [];
        this.output = output || undefined;
        this.exception = exception || undefined;
    }
    Object.defineProperty(JobStatus.prototype, "isExecuting", {
        get: function () { return this.state === 'Executing'; },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(JobStatus.prototype, "isClosed", {
        get: function () { return this.state === 'Closed'; },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(JobStatus.prototype, "errorResponse", {
        get: function () {
            if (this.exception) {
                this.exception.state = this.state;
            }
            else {
                this.exception = { 'state': this.state };
            }
            return this.exception;
        },
        enumerable: false,
        configurable: true
    });
    return JobStatus;
}());
exports.JobStatus = JobStatus;
