// @uipath/robot/dist/models/robotProcess.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RobotProcess = void 0;
var uipathRobot_1 = require("../uipathRobot");
var models_1 = require("../models");
/**
 * Robot process model containing details of the robot process available in current system
 */
var RobotProcess = /** @class */ (function () {
    /**
     * Default constructor
     * @param id Unique id of the robot process
     * @param name Robot Process name
     * @param description Process Description
     */
    function RobotProcess(id, name, description) {
        var _this = this;
        if (description === void 0) { description = undefined; }
        this.id = id;
        this.name = name;
        this.description = description;
        /**
         * Start the robot process
         * @param inArguments JSON object. In arguments to be passed to the robot if any.
         */
        this.start = function (inArguments) { return new models_1.JobPromise(new models_1.Job(_this.id, inArguments)); };
        /**
         * Install the robot process
         */
        this.install = function () { return uipathRobot_1.UiPathRobot.installProcess(_this.id); };
    }
    return RobotProcess;
}());
exports.RobotProcess = RobotProcess;
